### PR TITLE
Android ws/create list

### PR DESCRIPTION
**Pull request by valllllll2000**
Added the ListActivity to display list of launches.

Screenshots:
screenshot_1527088543
screenshot_1527092836

### CODE REVIEW

File: Android Workshop/app/src/main/java/com/womenwhocode/android/workshop/spacex/presentation/launches/LaunchesListPresenter.kt [hunk 1]
```diff
@@ -0,0 +1,45 @@
+package com.womenwhocode.android.workshop.spacex.presentation.launches
+
+import android.util.Log
+import com.womenwhocode.android.workshop.spacex.domain.model.Launch
+import com.womenwhocode.android.workshop.spacex.domain.usecases.GetLaunchesUseCase
+
+
+class LaunchesListPresenter(private val useCase: GetLaunchesUseCase) {
+
+    var view: LaunchesListActivity? = null
```
Comment (Perrankana) [inline Android Workshop/app/src/main/java/com/womenwhocode/android/workshop/spacex/presentation/launches/LaunchesListPresenter.kt]: Should it be LaunchesListView?

File: Android Workshop/app/src/main/java/com/womenwhocode/android/workshop/spacex/presentation/launches/LaunchesListPresenter.kt [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        view?.showErrorGettingLaunches()
+    }
+
+    private fun doOnGetLaunchesSuccess(launches: List<Launch>?) {
+        view?.hideLoading()
+        view?.displayLaunches(toViewLaunches(launches))
+    }
+
+    private fun toViewLaunches(launches: List<Launch>?): List<ViewLaunch> {
+        return launches?.map { toViewLaunch(it) } ?: ArrayList()
```
Comment (Perrankana) [inline Android Workshop/app/src/main/java/com/womenwhocode/android/workshop/spacex/presentation/launches/LaunchesListPresenter.kt]: I think you can replace "ArrayList()" by "emptyList()"

File: Android Workshop/app/src/main/java/com/womenwhocode/android/workshop/spacex/presentation/launches/LaunchesListPresenter.kt (hunk 1 above)
Comment (valllllll2000) [reply Android Workshop/app/src/main/java/com/womenwhocode/android/workshop/spacex/presentation/launches/LaunchesListPresenter.kt]: Oh you're so right, I mixed up!

File: Android Workshop/app/src/main/java/com/womenwhocode/android/workshop/spacex/presentation/launches/LaunchesListPresenter.kt (hunk 2 above)
Comment (valllllll2000) [reply Android Workshop/app/src/main/java/com/womenwhocode/android/workshop/spacex/presentation/launches/LaunchesListPresenter.kt]: Ok, changed!

File: Android Workshop/app/src/main/java/com/womenwhocode/android/workshop/spacex/presentation/launches/SpaceXAdapter.kt [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    override fun onBindViewHolder(holder: SpaceXViewHolder, position: Int) {
+        GlideApp.with(holder.nameTv.context).load(launches[position].smallImage).into(holder.smallIv)
+        holder.nameTv.text = launches[position].missionName
+        holder.yearTv.text = launches[position].launchYear
+        val color = if (launches[position].launchSuccess) Color.GREEN else Color.RED
+        holder.yearTv.setTextColor(color)
+        holder.locationTv.text = launches[position].siteName
+    }
+
+    override fun getItemCount(): Int {
```
Comment (Perrankana) [inline Android Workshop/app/src/main/java/com/womenwhocode/android/workshop/spacex/presentation/launches/SpaceXAdapter.kt]: you can do here:
`override fun getItemCount(): Int = launches.size`
same for 
`fun getItem(position: Int): ViewLaunch = launches[position]`

Comment (valllllll2000) [reply Android Workshop/app/src/main/java/com/womenwhocode/android/workshop/spacex/presentation/launches/SpaceXAdapter.kt]: Done!

**Pull request merged**